os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -211,8 +234,12 @@ int se_ameba_hal_set_key(hal_key_type mode, uint32_t key_idx, hal_data *key, hal
 	int ret;
 	inout_struc pub_prikey;
 
-	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
-		return HAL_INVALID_SLOT_RANGE;
+	ret = se_ameba_index_chk(key_idx);
+	if (HAL_SUCCESS != ret) {
+		return ret;
+	}
+	if (key_idx < FACTORY_KEY_INDEX_MAX) {	/* Factory Key Slot */
```

Comment:
you can move this condition before calling se_ameba_index_chk ..

Reply:
You can remove this because it is checked in se_ameba_index_chk.

Reply:
All key index check will be done in se_ameba_index_chk().